src/core/internal/vararg/sysv_x64.d

```diff
@@ -0,0 +1,304 @@
+                else
+                {
+                    *p1 = *cast(T1*) ap.stack_args;
+                    ap.stack_args += 8;
+                    *p2 = *cast(T2*) ap.stack_args;
+                    ap.stack_args += T2.sizeof.alignUp;
+                }
+            }
+            else // both in regular registers
+            {
+                void* a = void;
+                if (ap.offset_regs < 5 * 8 && T1.sizeof <= 8 && T2.sizeof <= 8)
+                {
+                    *p1 = *cast(T1*) (ap.reg_args + ap.offset_regs);
+                    ap.offset_regs += 8;
+                    a = ap.reg_args + ap.offset_regs;
+                    ap.offset_regs += 8;
+                }
+                else
+                {
+                    *p1 = *cast(T1*) ap.stack_args;
+                    ap.stack_args += 8;
+                    a = ap.stack_args;
+                    ap.stack_args += 8;
+                }
+                // Be careful not to go past the size of the actual argument
+                const sz2 = T.sizeof - 8;
+                (cast(void*) p2)[0..sz2] = a[0..sz2];
+            }
+
+            return result;
+        }
+        else
+        {
+            static assert(false);
+        }
+    }
+    else
+    {
+        static assert(false, "not a valid argument type for va_arg");
+    }
+}
+
+///
+void va_arg()(va_list ap, TypeInfo ti, void* parmn)
+{
+    TypeInfo arg1, arg2;
+    if (!ti.argTypes(arg1, arg2))
+    {
+        bool inXMMregister(TypeInfo arg) pure nothrow @safe
+        {
+            return (arg.flags & 2) != 0;
+        }
+
+        TypeInfo_Vector v1 = arg1 ? cast(TypeInfo_Vector) arg1 : null;
+        if (arg1 && (arg1.tsize <= 8 || v1))
+        {   // Arg is passed in one register
+            auto tsize = arg1.tsize;
+            void* p;
+            bool stack = false;
+            auto offset_fpregs_save = ap.offset_fpregs;
+            auto offset_regs_save = ap.offset_regs;
+        L1:
+            if (inXMMregister(arg1) || v1)
+            {   // Passed in XMM register
+                if (ap.offset_fpregs < (6 * 8 + 16 * 8) && !stack)
+                {
+                    p = ap.reg_args + ap.offset_fpregs;
+                    ap.offset_fpregs += 16;
+                }
+                else
+                {
+                    p = ap.stack_args;
+                    ap.stack_args += tsize.alignUp;
+                    stack = true;
+                }
+            }
+            else
+            {   // Passed in regular register
+                if (ap.offset_regs < 6 * 8 && !stack)
+                {
+                    p = ap.reg_args + ap.offset_regs;
+                    ap.offset_regs += 8;
+                }
+                else
+                {
+                    p = ap.stack_args;
+                    ap.stack_args += 8;
+                    stack = true;
+                }
+            }
+            parmn[0..tsize] = p[0..tsize];
+
+            if (arg2)
```

Comment:
then this could become 
```
if (!arg2)
    return;
```

Reply:
I wasn't trying to refactor/touch this x64 implementation at all. Especially not for these kind of aesthetics.

Reply:
Yeah, don't break one of the Walter rules, changing code whilst at the same time moving it.

Improvements can go in a follow up pr